### DISCUSSION TITLE
History of type-graphql.

### DISCUSSION THREAD

**Discussion by valerii15298** [MichalLytek/type-graphql#1287] [General]
Using this library I always wonder how people make great open source projects, what is the reason and motivation for creators and how open-source works in general. Would be interesting to hear about history of this library, how it was created, when and how this library grew.

I did not know that `type-graphql` was used by nest.js(though I noticed it is very similar to nest.js ecosystem) and in this context it is also interesting to hear why nest maintainers decided to diverge from `type-graphql`. So that there is now `nestjs/graphql` and `type-graphql`.

**Comment by itpropro** [comment]
I have no practical experience with Nest besides integrating into it in some projects, but as far as I understand the `nestjs/graphql` module is only easily usable within the Nest framework right?
The power of `type-graphql` is that is it framework agnostic as it's a GraphQL generator in the end, which is greatly shown in the prisma module [typegraphql-prisma](https://github.com/MichalLytek/typegraphql-prisma). 
There is obviously some better documentation and maintenance, as Nest is quite a big organization and `type-graphql` doesn't really have a team, it's mainly @MichalLytek maintaining it on his own (so leave him some sponsoring so he can invest more time 😉).
